src/index.ts

```diff
@@ -143,7 +143,7 @@ export const parser = (html: string, options: Options = {}): Node[] => {
     const buf: Node | undefined = bufArray.pop();
 
     if (buf && typeof buf === 'object' && buf.location && parser.endIndex !== null) {
-      buf.location.end = locationTracker.getPosition(parser.endIndex);
+      buf.location.end = locationTracker.getPosition((!parser.tagname || parser.tagname === buf.tag) ? parser.endIndex : parser.startIndex - 1);
```

Comment:
At the moment I can't see a way to know if the closing tag is implied or not. It seems to be possible to determine it with `tagname`... but it's private.

Reply:
The only other way I can think of is to see if an open tag has the same indices as the last close tag and rewrite it.

Reply:
@fb55 Is there another way to know if the closing tag is implied?

Reply:
One easy way would be to see if the indices have changed between events. What complicates this is that there can also be implied opening tags. Eg. `</p>` without a paragraph on the stack will lead to a `<p>` tag to be opened.

As an aside: I have just published `htmlparser2@7.0.0`, which has a much saner way of calculating indices.

Reply:
Thanks. Hadn't thought about invalid HTML, makes this even more fun. 😅

I'll take a look at trying to keep track of them.